src/dreyfus_httpd.erl

```diff
@@ -256,7 +262,7 @@ parse_index_param("bookmark", Value) ->
 parse_index_param("sort", Value) ->
     [{sort, ?JSON_DECODE(Value)}];
 parse_index_param("limit", Value) ->
-    [{limit, parse_non_negative_int_param("limit", Value, "max_limit", "200")}];
+    [{limit, ?JSON_DECODE(Value)}];
```

Comment:
I don't think we should broaden this here.

Reply:
Reading that `parse_index_param/2` is for GET method and `parse_json_index_param/2` is for POST method. `limit` can be specified both in GET and POST method. For partitioned database, the allowed max is 2000 while non-partitioned database is 200. So I try to extract the specified limit here and validate in https://github.com/cloudant-labs/dreyfus/pull/35/files/edb5a513cbb6054f28069ec72d0481a1c419992e#diff-1ffd08cb305df4b1411fbfd64633c17bR456 and https://github.com/cloudant-labs/dreyfus/pull/35/files/edb5a513cbb6054f28069ec72d0481a1c419992e#diff-1ffd08cb305df4b1411fbfd64633c17bR460